ethereum/mist
# New SendTx and TxHistory windows

#### What does it do?
This PR refactors `sendTransactionConfirmationWindow` into a react component called `SendTx`.

It also introduces a new Transaction History window (component called `TxHistory`).

The `txs` array in the redux store is now persisted to disk (new deps: `redux-persist`, `electron-store`, and `redux-persist-electron-storage`).

#### How to test it

There are four primary tx types handled: send ether, send tokens, execute a contract function, and deploy a new contract. Run through these scenarios. Reference the [QA Checklist](https://github.com/ethereum/mist/wiki/QA-Checklist) for a more thorough guide.

Note that you can also dispatch any custom action in the Redux DevTools to simulate error states.

#### Screenshots

_Screenshot example sending an ERC20 token:_

<img width="601" alt="screen_shot_2018-08-09_at_11 58 41_am" src="https://user-images.githubusercontent.com/22116/44046853-9aea6078-9ee1-11e8-998b-21ade7dadc1f.png">

_Gas loading/warning/error UI:_
![](http://g.recordit.co/p0Sb9fdA6Y.gif)
